### PR TITLE
Topbar toggle UI updates v2

### PR DESCRIPTION
**Pull request by lattemilk**
Sorry for the large commit 😅 the scss change touched all the components.
# Changes:
* Migrated to scss for easier styling
* Updated styling with material theming
* Added 'top-bar' component
* Removed mat-toolbar from 'navigation' component
* Replaced href with routerLink
* Added toggling of 'navigation' sidebar
* Fixed map positioning
* npm install updated package.json

# Expectations:
- [ ] Toggle the sidebar by clicking the hamburger menu icon
- [ ] Clicking the map zoom buttons doesn't hide topbar
- [ ] Navigate to views

# UI Update: 
Screen Shot 2022-10-20 at 5 42 46 PM

### CODE REVIEW

**Comment by lattemilk** [comment]
I'll move the top-bar back into the navigation component to utilize drawer.toggle().

File: src/interface/src/styles.scss [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+//         A700 : #000000,
+//     )
+// );
+
+// Define the palettes for your theme using the Material Design palettes available in palette.scss
+// (imported above). For each palette, you can optionally specify a default, lighter, and darker
+// hue. Available color palettes: https://material.io/design/color/
+$planscape-frontend-primary: mat.define-palette(mat.$gray-palette, 800);
+$planscape-frontend-accent: mat.define-palette(mat.$pink-palette, A200, A100, A400);
+// $planscape-frontend-primary: mat-palette($primary-palette, 500, 200, 800);
```
Comment (riecke) [inline src/interface/src/styles.scss:86]: Should these two lines be removed?

File: src/interface/src/styles.scss [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+// Create the theme object. A theme consists of configurations for individual
+// theming systems such as "color" or "typography".
+$planscape-frontend-theme: mat.define-light-theme((
+  color: (
+    primary: $planscape-frontend-primary,
+    accent: $planscape-frontend-accent,
+    // warn: $planscape-frontend-warn,
+  )
+));
+// $planscape-frontend-theme: mat-light-theme(
```
Comment (riecke) [inline src/interface/src/styles.scss:102]: Ditto for these four lines?

File: src/interface/src/styles.scss [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+// The warn palette is optional (defaults to red).
+$planscape-frontend-warn: mat.define-palette(mat.$red-palette);
+
+// Create the theme object. A theme consists of configurations for individual
+// theming systems such as "color" or "typography".
+$planscape-frontend-theme: mat.define-light-theme((
+  color: (
+    primary: $planscape-frontend-primary,
+    accent: $planscape-frontend-accent,
+    // warn: $planscape-frontend-warn,
```
Comment (riecke) [inline src/interface/src/styles.scss:99]: Should this line be removed?

File: src/interface/src/styles.scss [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+//   $planscape-frontend-primary,
+//   $planscape-frontend-accent,
+//   // $planscape-frontend-warn,
+// );
+
+// Include theme styles for core and each component used in your app.
+// Alternatively, you can import and @include the theme mixins for each component
+// that you are using.
+@include mat.all-component-themes($planscape-frontend-theme);
+// @include angular-material-theme($planscape-frontend-theme);
```
Comment (riecke) [inline src/interface/src/styles.scss:112]: Ditto.

**Review by riecke — APPROVED** [review]

File: src/interface/src/styles.scss (hunk 1 above)
Comment (lattemilk) [reply src/interface/src/styles.scss]: Thanks! I missed these.

**Review by thequeenofspades — APPROVED** [review]
This looks so good!! Thanks!

**Pull request merged**